### DISCUSSION TITLE
error when i make

### DISCUSSION THREAD

**Discussion by CGH20171006** [idaholab/moose#21675] [Q&A General] [answered]
This error occurs after I use  make -j4, especially for apps in the modules folder (I don't get this error for apps I create myself),I guess it's because I changed some files in the modules/phase_field folder,
what should  i do right now, is there any easy way to overwrite the moose files in GitHub to my computer again.

the error is:
(moose) cgh@LAPTOP-8KAA4L1I:~/projects/moose/modules/phase_field$ make -j4
Linking Executable /home/cgh/projects/moose/modules/phase_field/phase_field-opt...
/home/cgh/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/cgh/projects/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.so: undefined reference to `non-virtual thunk to SideIntegralPostprocessor::getValue()'
/home/cgh/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/cgh/projects/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.so: undefined reference to `MooseVariableData<double>::vectorTagValue(unsigned int) const'
/home/cgh/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/cgh/projects/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.so: undefined reference to `non-virtual thunk to ElementIntegralPostprocessor::getValue()'
/home/cgh/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/cgh/projects/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.so: undefined reference to `MooseVariableData<double>::vectorTagDofValue(unsigned int) const'
/home/cgh/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/cgh/projects/moose/modules/tensor_mechanics/lib/libtensor_mechanics-opt.so: undefined reference to `non-virtual thunk to Material::resolveOptionalProperties()'
collect2: error: ld returned 1 exit status
make: *** [/home/cgh/projects/moose/framework/app.mk:416: /home/cgh/projects/moose/modules/phase_field/phase_field-opt] Error 1

(ps:I am a Chinese undergraduate student and my English is very poor, I am sorry!)

**Comment by CGH20171006** [comment, marked as answer]
i have been overwrite the moose file,and the error has gone.

**Reply by tangweinuaa** [reply]
the same error i meat, can you find the dir /home/cgh/mambaforge3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/10.3.0/.
And, how do you overwrite the moose file

**Reply by GiudGiud** [reply]
By overwrite the moose files they mean they deleted the whole folder and cloned it again.

I dont think this is the same error. Could you please make your own post regardless?

**Reply by tangweinuaa** [reply]
ok, i did it by running update conda --all
Further, there are some conflicts or differences between the updated and old moose, which may lead to the same error.

**Reply by GiudGiud** [reply]
Once you have updated packages and an updated moose, please let us know if you get errors make-ing moose and your application
